fs/walk.go

```diff
@@ -250,6 +250,15 @@ func (dt DirTree) Dirs() (dirNames []string) {
 	return dirNames
 }
 
+// Prune remove directories from a directory tree.
+func (dt DirTree) Prune(dirNames []string) {
+	// We do not need to do a recursion.
+	// GC will take care of deleting child entries.
```

Comment:
The child entries are still in the map though.  They will be GC-ed properly but I wonder if any uses of DirTree will mind...  I don't think they will at the moment.